src/components/markdown-editor/markdown-editor.tsx

```diff
@@ -254,3 +264,29 @@ export function MarkdownEditorClient({
     </>
   );
 }
+
+function usePromiseRenderMap(
+  content: string,
+  renderPromise: (content: string) => Promise<React.JSX.Element>
+) {
+  const [promiseRenderMap, setPromiseRenderMap] = React.useState<
+    Map<string, Promise<React.JSX.Element>>
+  >(new Map());
+  const [, startTransition] = React.useTransition();
+  const lastRenderPromise = promiseRenderMap.get(content) ?? null;
+
+  const addPromise = React.useCallback(
+    (newContent: string) => {
+      startTransition(() => {
+        if (!promiseRenderMap.has(newContent)) {
+          const newMap = new Map(promiseRenderMap);
+          newMap.set(newContent, renderPromise(newContent));
+          setPromiseRenderMap(newMap);
```

Comment:
Why creating new Map every time? Wouldn't `const promiseRenderMap = React.useRef(new Map())` be better? or there's a trade-off?

Reply:
Hm... We need to have access to the latest promise at render time (which can go out of sync if we used refs).

But you are right about recreating the map every time, I can instead put that in a ref and put `lastRenderPromise` in state. 🤔

Reply:
done

Reply:
FYI, I'm watching the repo to learn from you. when I ask, I ask for learning purpose not for correcting.

thanks

Reply:
About the tradeoff, recreating the map can be costly and it's better to avoid it, but in this case, it is not that big of a deal.

Reply:
Usually this becomes a problem if you have to recreate an enormous map with many, MANY keys (maybe around thousands or millions).